### PR TITLE
Implement HolyFire

### PR DESCRIPTION
**Pull request by cpaca**
I suspect there might be a better way than calling onCreatureSpawn to all custom-blocks in the entire world when a creature attempts to spawn.

I'm not sure, though.

### CODE REVIEW

**Comment by cpaca** [comment]
The main reason I check for a null block is because IntelliJ was complaining about block being nullable.

I don't think it's possible for block==null to happen if the action is RIGHT_CLICK_BLOCK.

**Comment by cpaca** [comment]
That refactor probably did something wrong, because I had to do it manually since IntelliJ wasn't doing it correctly. When it's time to test this, it's also worth double-checking Plow didn't break.

**Comment by cpaca** [comment]
Heads up: If you're looking at the docs, there's two different Campfire classes. block.Campfire and block.data.type.Campfire.

**Comment by dlee13** [comment]
We still need to change how it's activated

File: src/main/java/xyz/holocons/mc/holoitemsrevamp/item/SaintQuartzItem.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import java.util.List;
+
+public class SaintQuartzItem extends CustomItem {
+
+    private static final String name = "saint_quartz";
+    private static final Material material = Material.QUARTZ;
+    private static final Component displayName = Component.text("Saint Quartz", NamedTextColor.BLUE);
+    private static final List<Component> lore = List.of(
+        Component.text("Surf the waves")
+    );
```
Comment (dlee13) [inline src/main/java/xyz/holocons/mc/holoitemsrevamp/item/SaintQuartzItem.java:20]: Wrong lore

**Pull request merged**